pkg/apis/kosmos/v1alpha1/distributionpolicy_types.go

```diff
@@ -0,0 +1,117 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope="Cluster"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp",description="CreationTimestamp is a timestamp representing the server time when this object was created. It is not guaranteed to be set in happens-before order across separate operations. Clients may not set this value. It is represented in RFC3339 form and is in UTC."
+
+type DistributionPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// DistributionSpec represents the desired behavior of DistributionPolicy.
+	// +required
+	DistributionSpec DistributionSpec `json:"spec"`
+}
+
+// DistributionSpec represents the desired behavior of DistributionPolicy.
+type DistributionSpec struct {
+	// ResourceSelectors used to select resources and is required.
+	// +required
+	// +kubebuilder:validation:MinItems=1
+	ResourceSelectors []ResourceSelector `json:"resourceSelectors"`
+
+	// PolicyTerms represents the rule for select nodes to distribute resources.
+	// +required
+	PolicyTerms []PolicyTerm `json:"policyTerms,omitempty"`
+}
+
+// ResourceSelector the resources will be selected.
+type ResourceSelector struct {
+	// Namespace of the target resource.
+	// Default is empty, which means cluster level.
+	// +optional
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,3,opt,name=namespace"`
+
+	// Name of the target resource.
+	// Default is empty, which means selecting all resources.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// Filter resource by labelSelector
+	// If target resource name is not empty, labelSelector will be ignored.
+	// +optional
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// Name of the Policy.
+	// +required
+	PolicyName string `json:"policyName,omitempty"`
+}
+
+type NodeType string
+
+const (
+	// HOSTNODE represents only the host nodes
+	HOSTNODE NodeType = "host"
+
+	// LEAFNODE represents only the leaf nodes
+	LEAFNODE NodeType = "leaf"
+
+	//MIXNODE represents the host nodes and child nodes
+	MIXNODE NodeType = "mix"
+)
+
+type PolicyTerm struct {
+	// +required
+	Name string `json:"name,omitempty"`
+
```

Comment:
follow this
  https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
> follow this https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

yes,have done